test/OpenTelemetry.Exporter.OneCollector.Tests/CommonSchemaJsonSerializationStateTests.cs

```diff
@@ -106,6 +139,41 @@ public void AddExtensionAttributeDuplicatesTest()
             json);
     }
 
+    [Fact]
+    public void AddExtensionAttributeMixedFieldAndPropertyTest()
+    {
+        using var stream = new MemoryStream();
+        using var writer = new Utf8JsonWriter(stream);
+
+        var state = new CommonSchemaJsonSerializationState("Test", writer);
+
+        // Note: For this test we expect the second entry to be dropped because
+        // the extension goes into "field tracking" mode
+        state.AddExtensionAttribute(new KeyValuePair<string, object?>("ext.foo.field1", 1));
+        state.AddExtensionAttribute(new KeyValuePair<string, object?>("ext.foo", "Hello world"));
+        state.AddExtensionAttribute(new KeyValuePair<string, object?>("ext.foo.field2", 2));
```

Comment:
Consider changing the behavior to avoid silent side-effects, perhaps align with what JavaScript is doing.

        state.AddExtensionAttribute(new KeyValuePair<string, object?>("ext.foo.field1", 1)); // now ext.foo = {"field1": 1}
        state.AddExtensionAttribute(new KeyValuePair<string, object?>("ext.foo", "Hello world")); // now ext.foo = "Hello world"
        state.AddExtensionAttribute(new KeyValuePair<string, object?>("ext.foo.field2", 2)); // throw exception since "Hello world"["field2"] = 2 is an invalid operation

Reply:
@reyang I just pushed some changes. How the new version works is wholly-owned extensions (the new feature) are limited to maps (`IReadOnlyList` or `IEnumerable` of `KeyValuePair<string, object?>`s) but will be merged with any fields (existing feature) set before or after the map. We could do full JavaScript-y style but I think this is a good balance of features without a whole bunch of cost (complexity & perf). We could always add support for other primitives in the future if the need arises. 

Let me know what you think about the new version!